notification.c

```diff
@@ -129,12 +130,14 @@ void close_notification(struct mako_notification *notif,
 	destroy_timer(notif->timer);
 	notif->timer = NULL;
 
-	wl_list_insert(&notif->state->history, &notif->link);
-	while (wl_list_length(&notif->state->history) >
-		notif->state->config.max_history) {
-		struct mako_notification *n =
-			wl_container_of(notif->state->history.prev, n, link);
-		destroy_notification(n);
+	if (add_to_history) {
+		wl_list_insert(&notif->state->history, &notif->link);
+		while (wl_list_length(&notif->state->history) >
+			notif->state->config.max_history) {
+			struct mako_notification *n =
+				wl_container_of(notif->state->history.prev, n, link);
+			destroy_notification(n);
+		}
 	}
```

Comment:
We need to `destroy_notification(notif)` in the else branch to avoid memory leaks.